### PR TITLE
[kn-curl] feature: initial kn curl plugin

### PR DESCRIPTION
**Pull request by maximilien**
1. initial PR for kn-curl plugin
2. first BASH plugin
3. missing some features (see Next Steps in README) that will be solved in next iteration

### CODE REVIEW

File: plugins/kn-curl/README.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+```bash
+# curl the service named hello
+kn curl hello
+```
+
+### Next steps
+
+1. Remove Istio assumption or at least deal with cases of non-istio Knative deployments.
+2. Determine the domain and use that or default `example.com` if not set.
```
[the remark this answers is not in the document]
Comment (maximilien) [reply plugins/kn-curl/README.md]: Nice. That’s a great suggestion @daisy-ycguo. I’ll add that now. Thanks :)

File: plugins/kn-curl/README.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+3. Add e2e test for this BASH plugin.
```
Comment (mattmoor) [inline plugins/kn-curl/README.md]: Add trailing newline:
```suggestion
3. Add e2e test for this BASH plugin.

```

File: plugins/kn-curl/README.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+1. Your Knative installation uses Istio for its mesh networking.
+2. You follow the Dependencies listed below.
+3. You don't have a custom domain for your services.
+
+### Dependencies
+
+1. Get the latest [`kn`](https://github.com/knative/client) binary for your platform.
+2. Make sure the `kn-curl` BASH file is located in your `~/.config/kn/plugins` directory. Of you are using a custom plugins repository, then add it there.
+3. Make sure the `kn-curl` BASH file has executable priviledges. You can achieve this with `chmod +x kn-curl`
```
Comment (mattmoor) [inline plugins/kn-curl/README.md]: Format markdown:
```suggestion
1. Get the latest [`kn`](https://github.com/knative/client) binary for your
   platform.
2. Make sure the `kn-curl` BASH file is located in your `~/.config/kn/plugins`
   directory. Of you are using a custom plugins repository, then add it there.
3. Make sure the `kn-curl` BASH file has executable priviledges. You can achieve
   this with `chmod +x kn-curl`
```

File: plugins/kn-curl/README.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+### Examples
+
+```bash
+# curl the service named hello
+kn curl hello
+```
+
+### Next steps
+
+1. Remove Istio assumption or at least deal with cases of non-istio Knative deployments.
```
Comment (mattmoor) [inline plugins/kn-curl/README.md]: Format markdown:
```suggestion
1. Remove Istio assumption or at least deal with cases of non-istio Knative
   deployments.
```

File: plugins/kn-curl/README.md (hunk 1 above)
Comment (mattmoor) [inline plugins/kn-curl/README.md]: Format markdown:
```suggestion
2. Determine the domain and use that or default `example.com` if not set.
3. Add e2e test for this BASH plugin.
```

File: plugins/kn-curl/README.md (hunk 2 above)
Comment (mattmoor) [inline plugins/kn-curl/README.md:28]: Add trailing newline:
```suggestion
3. Add e2e test for this BASH plugin.

```

File: plugins/kn-curl/README.md (hunk 3 above)
Comment (mattmoor) [inline plugins/kn-curl/README.md:15]: Format markdown:
```suggestion
1. Get the latest [`kn`](https://github.com/knative/client) binary for your
   platform.
2. Make sure the `kn-curl` BASH file is located in your `~/.config/kn/plugins`
   directory. Of you are using a custom plugins repository, then add it there.
3. Make sure the `kn-curl` BASH file has executable priviledges. You can achieve
   this with `chmod +x kn-curl`
```

File: plugins/kn-curl/README.md (hunk 4 above)
Comment (mattmoor) [inline plugins/kn-curl/README.md:26]: Format markdown:
```suggestion
1. Remove Istio assumption or at least deal with cases of non-istio Knative
   deployments.
```

File: plugins/kn-curl/README.md (hunk 1 above)
Comment (mattmoor) [inline plugins/kn-curl/README.md:27]: Format markdown:
```suggestion
2. Determine the domain and use that or default `example.com` if not set.
3. Add e2e test for this BASH plugin.
```

**Comment by maximilien** [comment]
*Update:* Finally getting back to this. I am planning to address all feedback this week as well as updated README.md

**Comment by maximilien** [comment]
Closing this. Will be submitting a better and different one. This was submitted before we had the recommended structure for plugins

**Pull request closed without merging**